### PR TITLE
Port #49480 to master

### PR DESCRIPTION
**Pull request by DmitryKuzmenko**
Port #49480 to master

Salt-based supportconfig. Re-implementation of Support Config Tool in
Salt.

### CODE REVIEW

File: salt/cli/support/collector.py [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+        :return:
+        """
+        try:
+            ret = self._get_runner(run_conf).run()
+        except SystemExit:
+            ret = "Runner is not available at this moment"
+            self.out.error(ret)
+        except Exception as ex:  # pylint: disable=broad-except
+            ret = "Unhandled exception occurred: {}".format(ex)
+            log.debug(ex, exc_info=True)
```
Comment (s0undt3ch) [inline salt/cli/support/collector.py:257]: ret? And not `ex`?

File: salt/cli/support/collector.py [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+            elif self.config["support_show_units"]:
+                self.out.put("List of available units:")
+                for idx, unit in enumerate(self.find_existing_configs(None)):
+                    msg_template = "  {}. ".format(idx + 1) + "{}"
+                    self.out.highlight(msg_template, unit)
+                exit_code = salt.defaults.exitcodes.EX_OK
+            else:
+                if not self.config["support_profile"]:
+                    self.print_help()
+                    raise SystemExit()
```
Comment (s0undt3ch) [inline salt/cli/support/collector.py:503]: Why not `self.exit()`?

File: tests/pytests/unit/cli/test_support.py [hunk 3]
```diff
@@ -0,0 +1,547 @@
...
+import salt.utils.files
+import yaml
+from salt.cli.support.collector import SaltSupport, SupportDataCollector
+from salt.cli.support.console import IndentOutput
+from salt.utils.color import get_colors
+from salt.utils.stringutils import to_bytes
+from tests.support.mock import MagicMock, patch
+
+
+class SaltSupportIndentOutputTest:
```
Comment (s0undt3ch) [inline tests/pytests/unit/cli/test_support.py:20]: Prefix the class with `Test`, `TestSaltSupportIndentOutput`

File: tests/pytests/unit/cli/test_support.py [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+    """
+    Unit Tests for the salt-support indent output.
+    """
+
+    def setUp(self):
```
Comment (s0undt3ch) [inline tests/pytests/unit/cli/test_support.py:25]: This does not work under PyTest, use fixtures.

File: salt/cli/support/collector.py (hunk 1 above)
Comment (DmitryKuzmenko) [reply salt/cli/support/collector.py:257]: The purpose is to collect all events so it just puts the error as well.

File: salt/cli/support/collector.py (hunk 2 above)
Comment (DmitryKuzmenko) [reply salt/cli/support/collector.py:503]: I've checked other cli tools and they use `SystemExit` as well as `self.exit()` in almost 50/50 cases. I'm not sure why.

File: salt/cli/support/collector.py (hunk 1 above)
Comment (s0undt3ch) [reply salt/cli/support/collector.py:257]: Sure, but the above `ret` is not used. I guess it was supposed to be on the log call, of not, nuke it.

File: salt/cli/support/collector.py (hunk 2 above)
Comment (s0undt3ch) [reply salt/cli/support/collector.py:503]: Ok.

File: salt/cli/support/collector.py (hunk 1 above)
Comment (DmitryKuzmenko) [reply salt/cli/support/collector.py:257]: Ah, I got you. Sorry.

**Comment by DmitryKuzmenko** [comment]
Superseded by https://github.com/saltstack/salt/pull/57054

**Pull request closed without merging**